src/sqlite.rs

```diff
@@ -442,4 +450,77 @@ impl Sqlite {
 
         Ok(())
     }
+
+    async fn get_indexes(
+        &self,
+        sqlite_conn: &mut SqliteConnection,
+    ) -> DataFusionResult<HashSet<String>> {
+        let query_result = sqlite_conn
+            .query_arrow(
+                format!("PRAGMA index_list({name})", name = self.table_name).as_str(),
+                &[],
+                None,
+            )
+            .await?;
+
+        let mut indexes = HashSet::new();
+
+        query_result
+            .try_collect::<Vec<RecordBatch>>()
+            .await
+            .into_iter()
+            .flatten()
+            .for_each(|batch| {
+                if let Some(name_array) = batch
+                    .column_by_name("name")
+                    .and_then(|col| col.as_any().downcast_ref::<StringArray>())
+                {
+                    for index_name in name_array.iter().flatten() {
+                        // Filter out SQLite's auto-generated indexes
+                        if !index_name.starts_with("sqlite_autoindex_") {
+                            indexes.insert(index_name.to_string());
+                        }
+                    }
+                }
+            });
+
+        Ok(indexes)
+    }
+
+    async fn verify_indexes_match(
+        &self,
+        sqlite_conn: &mut SqliteConnection,
+        indexes: &[(ColumnReference, IndexType)],
+    ) -> DataFusionResult<bool> {
+        let expected_indexes_str_map: HashSet<String> = indexes
+            .iter()
+            .map(|(col, _)| IndexBuilder::new(&self.table_name, col.iter().collect()).index_name())
+            .collect();
+
+        let actual_indexes_str_map = self.get_indexes(sqlite_conn).await?;
+
+        let missing_in_actual = expected_indexes_str_map
+            .difference(&actual_indexes_str_map)
+            .collect::<Vec<_>>();
+        let extra_in_actual = actual_indexes_str_map
+            .difference(&expected_indexes_str_map)
+            .collect::<Vec<_>>();
+
+        if !missing_in_actual.is_empty() {
+            tracing::warn!(
```

Comment:
Not sure we should warn here, we might just want to return back detected issues.

Reply:
The warning should probably just explicitly state index instead of `schema mismatch`. Index is not part of the table schema in SQLite, instead they are separate database object

Reply:
Updated - thank you